docs/sources/setup-grafana/configure-docker.md

```diff
@@ -57,31 +82,150 @@ The following settings are hard-coded when launching the Grafana Docker containe
 
-> Only available in Grafana v5.2 and later.
+Example:
+
+The below example shows how to build the customized Grafana Docker Image and provides the Image Render Plugin in the build argument.
 
-It's possible to supply Grafana with configuration through files. This works well with [Docker Secrets](https://docs.docker.com/engine/swarm/secrets/) as the secrets by default gets mapped into `/run/secrets/<name of secret>` of the container.
+```bash
+#go to the folder
+cd packaging/docker/custom
 
-You can do this with any of the configuration options in conf/grafana.ini by setting `GF_<SectionName>_<KeyName>__FILE` to the path of the file holding the secret.
+#running the build command
+docker build \
+--build-arg "GRAFANA_VERSION=latest" \
+--build-arg "GF_INSTALL_IMAGE_RENDERER_PLUGIN=true" \
+-t grafana-custom .
+
+# running the docker run command
+docker run -d -p 3000:3000 --name=grafana grafana-custom
+```
+
+# Logging
+
+Logs in the Docker container go to `STDOUT` by default, as is common in the Docker world. Change this by setting a different [log mode](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#mode). Available options are `console`, `file`, and `syslog`. The default is `console` and `file`. Use spaces to separate multiple modes, e.g `console file`
+
+Example:
+
+To change the log mode, use the `GF_LOG_MODE` environment variable and define your mode e.g. to use `console file`, run the command:
+
+```bash
+# Run Grafana while logging to both standard out 
+# and /var/log/grafana/grafana.log
+
+docker run -p 3000:3000 -e "GF_LOG_MODE=console file" grafana/grafana-enterprise
+```
+# Configure Grafana with Docker Secrets
+
+You can supply Grafana sensitive information e.g. secrets, login credentials etc. with configuration through files. This works well with [Docker Secrets](https://docs.docker.com/engine/swarm/secrets/) as the secrets by default are mapped into the following location in the container: `/run/secrets/<name of secret>.
+` 
+You can do this with any of the configuration options in `conf/grafana.ini` by setting `GF_<SectionName>_<KeyName>__FILE` to the path of the file holding the secret.
 
 For example, you could set the admin password this way:
 
 - Admin password secret: `/run/secrets/admin_password`
 - Environment variable: `GF_SECURITY_ADMIN_PASSWORD__FILE=/run/secrets/admin_password`
+# Configure AWS credentials for CloudWatch Support
 
-## Configure AWS credentials for CloudWatch Support
+Defining secrets is very important on the public cloud so that it is not visible as plain text and provide a layer of security.
 
 ```bash
```

Comment:
none of these example commands have any context. Can you pls add some information about what's going on here?

Reply:
rewrote this section in more better way. please check

Reply:
thank you. reviewing!